### PR TITLE
Refactor Vagrantfile.

### PR DESCRIPTION
**Pull request by ehelms**
This refactors out the box processing logic into it's own testable
class. Further, the boxes are moved into a configuration file and
loaded/defined the same as custom boxes were loaded and defined
previously.

This does introduce one "api" change in that the previous "nightly"
boxes are renamed to "centos6-nightly" and "centos7-nightly". This
also means that the base centos6 and centos7 boxes (those without any
ties to install scripts etc.) are available as first class citizens.
In other words, you can vagrant up a bare bones centos6 or centos7 box
directly now.

### CODE REVIEW

**Comment by stbenjam** [comment]
@ehelms This one needs rebasing too

**Comment by ehelms** [comment]
Updated
